### PR TITLE
More SQLite testing

### PR DESCRIPTION
**Pull request by natemcmaster**
Depends on https://github.com/aspnet/Microsoft.Data.Sqlite/pull/84 for the transaction tests to work correctly.

TODO: DbConnection is not always disposed correctly--> test *.db files are left behind.

### CODE REVIEW

**Comment by anpete** [comment]
:ship: :it:

File: test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs [hunk 1]
```diff
@@ -88,29 +102,30 @@ private DbCommand CreateCommand(string commandText, object[] parameters)
...
 
-        public static string CreateConnectionString(string name) =>
+        public static string CreateConnectionString(string name, bool sharedCache = false) =>
             new SqliteConnectionStringBuilder
-                {
-                    DataSource = name + ".db"
-                }
+            {
+                DataSource = name + ".db",
+                SharedCacheMode = sharedCache
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs]: Any reason not to just always use a shared cache?

File: test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs [hunk 2]
```diff
@@ -17,8 +18,17 @@ public class SqliteTestStore : RelationalTestStore
...
-        public static SqliteTestStore CreateScratch() => 
-            new SqliteTestStore("scratch-" + Interlocked.Increment(ref _scratchCount)).CreateTransient();
+        public static SqliteTestStore CreateScratch(bool sharedCache = false)
+        {
+            string name;
+            do
+            {
+                name = "scratch-" + Interlocked.Increment(ref _scratchCount);
+            }
+            while (File.Exists(name + ".db"));
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs]: With this, you can re-add the built-in types tests.

**Comment by bricelam** [comment]
:shipit:

File: test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs (hunk 1 above)
Comment (natemcmaster) [inline test/EntityFramework.Sqlite.FunctionalTests/SqliteTestStore.cs]: I can only think of one reason: it is not the default SQLite behavior.

**Pull request merged**